### ISSUE TITLE
Accessibility: calcite-dropdown groups headings are not read by NVDA and VoiceOver screen readers

### ISSUE THREAD

**Issue by MikeTschudi** [Esri/calcite-components#674]
### Summary

The NVDA and VoiceOver screen reader users do not receive `dropdown-group` context when navigating through `dropdown-item`s.

### Actual Behavior

On the [Groups and selection modes example](https://esri.github.io/calcite-components/?path=/story/dropdown--groups-and-selection-modes) page, as one tabs in the dropdown, one hears:
* "menu"
* "orange two of eight"
* "grape three of eight"
* "aspargus four of eight"
* "potato five of eight"
* "yam six of eight"
* "plant beans seven of eight"
* "add peas eight of eight"

The group titles provide useful context.

### Expected Behavior

- Read back the group name to each `dropdown-item`, providing useful context back to the user.
- Have consistent experiences across screen readers (JAWS, NVDA, and VoiceOver)

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/dropdown/#sample

### Reproduction Steps

1. Enable NVDA on Windows 10 -**OR**- VoiceOver on Mac OS
2. Navigate to the [sample](https://developers.arcgis.com/calcite-design-system/components/dropdown/#sample)
3. With the screen reader active, navigate via keyboard to focus on the "Activate dropdown" button to trigger the dropdown.
4. Listen when interacting with the `dropdown-item`s residing in `dropdown-group`s. The item is announced, as is its placement in the group, however the group's name is not announced. _Note: the group name **is announced** on JAWS._

### Reproduction Version

beta.83

### Relevant Info

- [x] Windows (NVDA Firefox)
- [x] Mac OS (Safari/VoiceOver)
- [x] ~Windows (JAWS Chrome and Firefox)~ - group context provided

**Comment by macandcheese** [comment]
Thanks for reporting. Would a better experience to have each group have its own "count" (3/3/2 vs. 8)?

**Comment by MikeTschudi** [comment]
I believe that it would be a better experience because the groups can be different, such as the example's "Select one", "Select multi", and "Select none...".

It must already be available, however, because we get that with the use of the dropdown in the solution deployment app (reference [above](https://github.com/ArcGIS/solution-deployment-app/issues/316)): the first group is "Sort by" and the second group is "Sort direction" with counts 3 & 2.

**Comment by MikeTschudi** [comment]
At calcite-components version 1.0.0-beta.46, the latest on [unpkg.com](https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.esm.js), continues to be an issue.

```html
<!doctype html>
<html lang="en" dir="ltr">
<head>
  <link rel="stylesheet" type="text/css" href="https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.css" />
  <script type="module" src="https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.esm.js"></script>
  <script nomodule="" src="https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.js"></script>
</head>
<body>
  <!-- https://esri.github.io/calcite-components/?path=/story/components-dropdown--groups-and-selection-modes -->
  <calcite-dropdown>
    <calcite-button slot="dropdown-trigger">Open Dropdown</calcite-button>
    <calcite-dropdown-group group-title="Select one">
      <calcite-dropdown-item>Apple</calcite-dropdown-item>
      <calcite-dropdown-item active>Orange</calcite-dropdown-item>
      <calcite-dropdown-item>Grape</calcite-dropdown-item>
    </calcite-dropdown-group>
    <calcite-dropdown-group group-title="Select multi" selection-mode="multi">
      <calcite-dropdown-item>Asparagus</calcite-dropdown-item>
      <calcite-dropdown-item active>Potato</calcite-dropdown-item>
      <calcite-dropdown-item>Yam</calcite-dropdown-item>
    </calcite-dropdown-group>
    <calcite-dropdown-group group-title="Select none (useful for actions)" selection-mode="none">
      <calcite-dropdown-item>Plant beans</calcite-dropdown-item>
      <calcite-dropdown-item active>Add peas</calcite-dropdown-item>
    </calcite-dropdown-group>
  </calcite-dropdown>
</body>
</html>
```

**Comment by julio8a** [comment]
Still valid

**Comment by shoe913** [comment]
Still valid, @cosbyr, can we re-open?

**Comment by MikeTschudi** [comment]
@1.0.0-beta.57 & NVDA 2020.4

The menu
![image](https://user-images.githubusercontent.com/2125181/123005682-57a00b80-d36b-11eb-9b40-18da1765bcfc.png)

is read as
```
clickable  Open Dropdown  button  
radio menu item  checked  1 of 3
radio menu item  not checked  2 of 3
radio menu item  not checked  3 of 3
check menu item  checked  1 of 3
check menu item  checked  2 of 3
check menu item  not checked  3 of 3
menu item  1 of 2
menu item  2 of 2
```

**Comment by MikeTschudi** [comment]
@1.0.0-beta.62 & NVDA 2020.4, continues to be a problem

**Comment by MikeTschudi** [comment]
@1.0.0-beta.66 & NVDA 2021.2, menu items are read but group headings are still not seen. Earlier example is now read as
```
clickable  Open Dropdown  button  
Orange  radio menu item  checked  2 of 3
menu
Grape  radio menu item  not checked  3 of 3
Asparagus  not checked  1 of 3
Potato  checked  2 of 3
Yam  not checked  3 of 3
Plant beans  1 of 2
Add peas  2 of 2
Apple
```

**Comment by MikeTschudi** [comment]
@1.0.0-beta.68 & NVDA 2021.2, same as https://github.com/Esri/calcite-components/issues/674#issuecomment-932269008: menu items are read but group headings are not read.

There seems to be an error in NVDA: when one types Enter on the Open Dropdown button, one hears only "menu" but one sees in the NVDA log
```
Orange  radio menu item  checked  2 of 3
menu
```
One only hears "Orange..." if one arrows away and then back.

**Comment by macandcheese** [comment]
Keep open - @benelan maybe we can re-triage this one?

**Comment by macandcheese** [comment]
Keep open

**Comment by MikeTschudi** [comment]
At NVDA 2021.3.1 & calcite-components version 1.0.0-beta.76, the latest on [unpkg.com](https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.esm.js), continues to be an issue.

```
clickable  Open Dropdown  button 
Orange  radio menu item  checked  2 of 3
Grape  radio menu item  not checked  3 of 3
Asparagus  not checked  1 of 3
Potato  checked  2 of 3
Yam  not checked  3 of 3
Plant beans  1 of 2
Add peas  2 of 2
```

**Comment by MikeTschudi** [comment]
https://codepen.io/miketschudi/pen/dyZZjRG

**Comment by jcfranco** [comment]
Our components will go over an a11y audit. Blocking this until that is completed.

**Comment by geospatialem** [comment]
Updating the issue context to focus on NVDA and VoiceOver. In JAWS, the group name context is read back in both the Firefox and Chrome browsers. 

Since [JAWS is used by just over 50% of screen reader users](https://webaim.org/projects/screenreadersurvey9/#primary), downgrading the priority of this to `medium`, but as @jcfranco, would be good to wait on this until our audit takes place in the coming weeks.

**Comment by geospatialem** [comment]
Context for the group is provided across JAWS, NVDA, and Voiceover. Closing this issue for the resolution.
